src/common/misc.c

```diff
@@ -322,15 +322,17 @@ daos_sgl_get_bytes(d_sg_list_t *sgl, bool check_buf, struct daos_sgl_idx *idx,
 	daos_size_t len;
 
 	if (p_buf_len != NULL)
-	*p_buf_len = 0;
+		*p_buf_len = 0;
 
 	if (idx->iov_idx >= sgl->sg_nr)
 		return true; /** no data in sgl to get bytes from */
 
 	len = check_buf ? sgl->sg_iovs[idx->iov_idx].iov_buf_len :
 		sgl->sg_iovs[idx->iov_idx].iov_len;
 
-	D_ASSERT(idx->iov_offset < len);
+	D_ASSERT(idx->iov_offset <= len);
```

Comment:
I think the 'iov_offset' should never equal to 'len' since we'll move to next iov once "iov_offset == len", right? (see following code in this function)

Reply:
one possible case is that if user passed in sgl with one iov with zero length, then the idx->iov_offset possily == len == 0, it is extreme cornel case but seems possible.

Reply:
what's the use case of 0 length iov? I tend to think we'd assert iov length > 0 to avoid potential bug.